### PR TITLE
ENH more precise noarch for python

### PR DESCRIPTION
**Pull request by beckermr**
This PR computes noarch more precisely.

### CODE REVIEW

File: conda_forge_tick/migrators/migration_yaml.py [hunk 1]
```diff
@@ -412,6 +412,32 @@ def order(
         )
 
 
+def _all_noarch(attrs, only_python=False):
+    if not only_python:
+        all_noarch = "noarch" in attrs.get("meta_yaml", {}).get("build", {})
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators/migration_yaml.py:417]: I would make a var for `attrs.get("meta_yaml", {})` since this uses it a bunch.

**Comment by beckermr** [comment]
I ran pre commit locally but the output differs. This might be a thing with black.

**Comment by beckermr** [comment]
```
(base) clarence:cf-scripts beckermr$ pre-commit run -a
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
black....................................................................Passed
pyupgrade................................................................Passed
Add trailing commas......................................................Passed
codespell................................................................Passed
```

**Comment by beckermr** [comment]
@CJ-Wright IDK how to fix the pre commit thing, but I looked locally and this PR works for excluding llvmdev and clangdev

**Comment by mariusvniekerk** [comment]
I'll do an upgrade of the precommit stuff in a seperate PR

**Comment by beckermr** [comment]
The following nodes are removed from python 39

```
{'beautifulsoup4',
 'biorbd-casadi',
 'black',
 'brial',
 'clangdev',
 'evalml-core',
 'geoviews',
 'h5utils',
 'llvmdev',
 'lsst-documenteer',
 'ndarray',
 'nlp_primitives',
 'openpiv',
 'parsmurf',
 'prompt_toolkit',
 'property-cached',
 'pymedphys',
 'starlette',
 'typing_extensions'}
```

**Comment by mariusvniekerk** [comment]
LGTM!

**Pull request merged**